bin/fuel-core/src/cli/run/p2p.rs

```diff
@@ -321,8 +321,9 @@ impl P2PArgs {
             reserved_nodes: self.reserved_nodes,
             reserved_nodes_only_mode: self.reserved_nodes_only_mode,
             enable_mdns: self.enable_mdns,
-            max_peers_connected: self.max_peers_connected,
-            max_connections_per_peer: self.max_connections_per_peer,
+            max_discovery_peers_connected: self.max_peers_connected,
+            max_gossipsub_peers_connected: 100,
+            max_request_response_peers_connected: 100,
```

Comment:
Some suggestions on these two lines:

1. should these values be configurable, and also have a hard limit? 

e.g. something like 

```suggestion
            max_gossipsub_peers_connected: self.max_gossipsub_peers_connected.min(100),
            max_request_response_peers_connected: self.max_request_response_peers_connected.min(100),
```

2. Maybe best to declare the hardcoded constants as const variables? 

3. You renamed `self.max_peers_connected, but not `self.max_peers_connected`. (Ther argument has `--max-peers-connected` as argument name hardcoded in clap_derive, so there should be no risk in renaming this

Reply:
>3. You renamed self.max_peers_connected, but not self.max_peers_connected. (Ther argument has --max-peers-connected` as argument name hardcoded in clap_derive, so there should be no risk in renaming this

I'm not following this point. It looks like you mentioned the same thing twice `self.max_peers_connected`

Reply:
I did your first two recs, but my comment I added might be too vague. I don't completely understand why these two were hardcoded but the `discovery` limit wasn't.

Reply:
> I'm not following this point. It looks like you mentioned the same thing twice self.max_peers_connected 

sorry, I meant: you renamed `Config::max_peers_connected` into `Config::max_discovery_peers_connected`, while leaving the same name for `P2PArgs::max_peers_connected`.